t/lib/proxy-wasm-tests/hostcalls/Cargo.toml

```diff
@@ -8,7 +8,8 @@ edition = "2018"
 crate-type = ["cdylib"]
 
 [dependencies]
-proxy-wasm = "0.2"
+#proxy-wasm = "0.2"
+proxy-wasm = { git = "https://github.com/casimiro/proxy-wasm-rust-sdk.git", branch = "foreign-function-callback" }
```

Comment:
Yeah, I was hoping the changes in the upstream would be merged by the end of this PR.

On one hand, this workaround seems very subpar and makes me want to consider it a blocker.

On the other, that separate branch contains a single commit that's already been approved by the maintainers and should be part of the next release; and using this separate branch would allow us to ship foreign function support in time for the 3.9 release of the gateway.